src/types/dict.rs

```diff
@@ -175,6 +205,39 @@ impl PyDict {
         }
     }
 
+    /// Returns a view of dict keys.
+    ///
+    /// This is equivalent to the Python expression `dict.keys()`.
+    pub fn keys_view(&self) -> &PyDictKeys {
+        unsafe {
+            PyDictKeys::try_from_unchecked(
+                self.call_method0(intern!(self.py(), "keys")).unwrap(),
```

Comment:
Can `self` be a dict subclass -- which can therefore return a different type?

Reply:
Yes. I guess that would be the advantage of having a C API - then we could always hit the guaranteed pathway rather than lookup and allow subclasses to override things.